### DISCUSSION TITLE
Geoserver does not open after restart

### DISCUSSION THREAD

**Discussion by orttak** [kartoza/docker-geoserver#580] [Q&A]
Hi,
I use geoserver and postgis as I share below. After restarting Docker Compose, it gives geoserver error. Is there anyone who can help?
![Uploading image.png…]()

`version: "3.3"

volumes:
  tosca-db-data:
  tosca-geoserver-data-dir:

services:
  db:
    image: kartoza/postgis:15-3.3
    container_name: tosca-2-db
    volumes:
      - tosca-db-data:/var/lib/postgresql
    ports:
      - ${POSTGRES_PORT}:5432
    environment:
      - POSTGRES_DB=${POSTGRES_DB}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASS=${POSTGRES_PASS}
      - ALLOW_IP_RANGE=${ALLOW_IP_RANGE}
      - FORCE_SSL=FALSE
    restart: on-failure
    healthcheck:
      test: "PGPASSWORD=${POSTGRES_PASS} pg_isready -h 127.0.0.1 -U ${POSTGRES_USER} -d ${POSTGRES_DB}"

  geoserver:
    image: kartoza/geoserver:2.23.1
    container_name: tosca-2-geoserver
    volumes:
      - tosca-geoserver-data-dir:/opt/geoserver/data_dir
      # sample volume attachment. Please give full path.
      #- /home/tosca/sample_data:/mnt/data
    ports:
      - ${GEOSERVER_PORT}:8080
    restart: on-failure
    env_file:
      - .env
    depends_on:
      - db
`

error log
`User
20 Oct 13:12:06 INFO   [jdbc.JDBCQuotaStore] - Closing up the JDBC quota store 
20 Oct 13:12:06 WARN   [support.DisposableBeanAdapter] - Custom destroy method 'destroy' on bean with name 'gwcMemoryBlobStore' threw an exception: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@65fbde35[Not completed, task = org.geowebcache.storage.blobstore.memory.MemoryBlobStore$BlobStoreTask@476acb39] rejected from java.util.concurrent.ThreadPoolExecutor@3cd63294[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 1]
20 Oct 13:12:06 INFO   [geoserver.logging] - Beginning GeoServer cleanup sequence
root@d90fd948b939:/opt/geoserver/data_dir/logs# ^`

`20 Oct 13:12:05 CONFIG [wms.WMSService] - Will proxy requests to backend that are not getmap or getcapabilities.
20 Oct 13:12:05 INFO   [geoserver.monitor] - Monitor extension enabled
[Fatal Error] :-1:-1: Premature end of file.
20-Oct-2023 13:12:06.532 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal One or more listeners failed to start. Full details will be found in the appropriate container log file
20-Oct-2023 13:12:06.537 SEVERE [main] org.apache.catalina.core.StandardContext.startInternal Context [/geoserver] startup failed due to previous errors
20 Oct 13:12:06 CONFIG [gwc.layer] - GeoServer TileLayer store base directory is: gwc-layers
20 Oct 13:12:06 CONFIG [gwc.layer] - Loading tile layers from gwc-layers
20 Oct 13:12:06 CONFIG [gwc.layer] - Loaded 2 tile layers in 16.95 ms
20 Oct 13:12:06 INFO   [jdbc.JDBCQuotaStore] - Closing up the JDBC quota store 
20 Oct 13:12:06 WARN   [support.DisposableBeanAdapter] - Custom destroy method 'destroy' on bean with name 'gwcMemoryBlobStore' threw an exception: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@65fbde35[Not completed, task = org.geowebcache.storage.blobstore.memory.MemoryBlobStore$BlobStoreTask@476acb39] rejected from java.util.concurrent.ThreadPoolExecutor@3cd63294[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 1]
20 Oct 13:12:06 INFO   [geoserver.logging] - Beginning GeoServer cleanup sequence
20-Oct-2023 13:12:07.197 WARNING [main] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [geoserver] appears to have started a thread named [FileSystemWatcher-1] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.base@11.0.18/jdk.internal.misc.Unsafe.park(Native Method)
 java.base@11.0.18/java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:234)
 java.base@11.0.18/java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2123)
 java.base@11.0.18/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:1182)
 java.base@11.0.18/java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:899)
 java.base@11.0.18/java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1054)
 java.base@11.0.18/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1114)
 java.base@11.0.18/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
 java.base@11.0.18/java.lang.Thread.run(Thread.java:829)
20-Oct-2023 13:12:07.198 WARNING [main] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [geoserver] appears to have started a thread named [Timer-0] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.base@11.0.18/java.lang.Object.wait(Native Method)
 java.base@11.0.18/java.util.TimerThread.mainLoop(Timer.java:553)
 java.base@11.0.18/java.util.TimerThread.run(Timer.java:506)
20-Oct-2023 13:12:07.198 WARNING [main] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [geoserver] appears to have started a thread named [EMF Reference Cleaner] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.base@11.0.18/java.lang.Object.wait(Native Method)
 java.base@11.0.18/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:155)
 java.base@11.0.18/java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:176)
 org.eclipse.emf.common.util.CommonUtil$1ReferenceClearingQueuePollingThread.run(CommonUtil.java:70)
20-Oct-2023 13:12:07.199 WARNING [main] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [geoserver] appears to have started a thread named [H2 File Lock Watchdog /opt/geoserver/data_dir/gwc/diskquota_page_store_h2/diskquota.lock.db] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.base@11.0.18/java.lang.Thread.sleep(Native Method)
 org.h2.store.FileLock$1.run(FileLock.java:341)
 java.base@11.0.18/java.lang.Thread.run(Thread.java:829)
20-Oct-2023 13:12:07.200 SEVERE [main] org.apache.catalina.loader.WebappClassLoaderBase.checkThreadLocalMapForLeaks The web application [geoserver] created a ThreadLocal with key of type [java.lang.ThreadLocal.SuppliedThreadLocal] (value [java.lang.ThreadLocal$SuppliedThreadLocal@70ea1a57]) and a value of type [org.geowebcache.storage.CompositeBlobStore.StoreSuitabilityCheck] (value [EXISTING]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
20-Oct-2023 13:12:07.201 SEVERE [main] org.apache.catalina.loader.WebappClassLoaderBase.checkThreadLocalMapForLeaks The web application [geoserver] created a ThreadLocal with key of type [java.lang.ThreadLocal] (value [java.lang.ThreadLocal@7879f5d6]) and a value of type [org.springframework.security.core.context.SecurityContextImpl] (value [SecurityContextImpl [Null authentication]]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
20-Oct-2023 13:12:07.211 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/geoserver] has finished in [19,316] ms
20-Oct-2023 13:12:07.213 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]
20-Oct-2023 13:12:07.229 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [19390] milliseconds `

**Comment by orttak** [comment]
I just change the name of the volume name in my compose setting. After that I don't get any error after rebuild my env. I don't understand the issues:) I close this question

**Comment by NyakudyaA** [comment]
@orttak In production it is recommended to use https://github.com/kartoza/docker-geoserver#enable-disk-quota-storage-in-postgresql-backend. I suggest you always have that setup because the default uses H2 which can cause issue if it get's corrupted or something like that
